core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
The problem with the ConstReader is a minor implementation detail, imho, that can be easily circumnavigated. The current RequestReader implementation with all the copy-paste code is not a piece of beauty that needs to be preserved like the population of Panda bears. :-)

I cited three reasons against the `suchThat` syntax and in my eyes they have more weight than the few problems we'd need to overcome with the original idea. Do you have an elegant solution regarding 3) above for example?

Maybe we can get a few more opinions on that?